server/src/test/java/io/crate/analyze/ScalarsAndRefsToTrueTest.java

```diff
@@ -153,4 +156,26 @@ public void test_IsNull_and_IsNotNull_nested_with_complex_expression() {
             "(NOT(((table_name = 'foo') AND null) IS NULL)) OR (NOT(((table_name = 'foo') AND null) IS NOT NULL))");
         assertThat(symbol).isLiteral(true);
     }
+
+    /**
+     * Tracks a bug: {@see https://github.com/crate/crate/issues/16614}
+     * NOT should be just removed and the inner argument should be returned instead, if it's normalized to a literal
+     */
+    @Test
+    public void test_not_on_top_of_IsNotNull() {
```

Comment:
```suggestion
    @Test
    public void test_double_not_on_top_of_IsNotNull() {
```

Maybe we can also add single NOT on top of not null (should be also true). Basically https://github.com/crate/crate/pull/16650/files#diff-9255ffd7f793f59182b99c78d7cdd3cceb3b7ff9505533f6865efeac8e8df94fR163-R167

Reply:
I removed the extra NOT, doesn't add any more value.